Update uninstall guide involving systemd (#7324)
NixOS/nix: doc/manual/src/installation/installing-binary.md

### Linux

To begin, if you are on Linux with systemd, remove the Nix daemon service:

```console
sudo systemctl stop nix-daemon.service
sudo systemctl disable nix-daemon.socket nix-daemon.service
sudo systemctl daemon-reload
```

Then you can remove systemd service files:

```console
sudo rm /etc/systemd/system/nix-daemon.service /etc/systemd/system/nix-daemon.socket
```

> Also, the installer script uses systemd-tmpfiles (if presents) to create the socket directory. You may also want to remove the configuration for that:

Reviewer: ```suggestion
The installer script uses systemd-tmpfiles (if present) to create the socket directory.
You may also want to remove the configuration for that:
```

> To begin, if you are on Linux with systemd, remove the Nix daemon service:
> 
> ```console
> sudo systemctl stop nix-daemon.service
> sudo systemctl disable nix-daemon.socket nix-daemon.service
> sudo systemctl daemon-reload
> ```
> 
> Then you can remove systemd service files:
> 
> ```console
> sudo rm /etc/systemd/system/nix-daemon.service /etc/systemd/system/nix-daemon.socket
> ```
> 
> Also, the installer script uses systemd-tmpfiles (if presents) to create the socket directory. You may also want to remove the configuration for that:
> 
> ```console
> sudo rm /etc/tmpfiles.d/nix-daemon.conf
> ```

Reviewer: Since it's not absolutely clear from text or structure if the third instruction also only relates to systemd-based systems, what about we indent these into a list?

````suggestion
If you are on Linux with systemd:

1. Remove the Nix daemon service:

  ```console
  sudo systemctl stop nix-daemon.service
  sudo systemctl disable nix-daemon.socket nix-daemon.service
  sudo systemctl daemon-reload
  ```

  1. Remove systemd service files:

  ```console
  sudo rm /etc/systemd/system/nix-daemon.service /etc/systemd/system/nix-daemon.socket
  ```

  1. The installer script uses systemd-tmpfiles to create the socket directory.
     You may also want to remove the configuration for that:

  ```console
  sudo rm /etc/tmpfiles.d/nix-daemon.conf
  ```
````

Author: This makes it a lot clearer. Thanks!

### Linux

To begin, if you are on Linux with systemd, remove the Nix daemon service:

```console

> sudo systemctl stop nix-daemon.socket
> sudo systemctl stop nix-daemon.service
> sudo systemctl disable nix-daemon.socket
> sudo systemctl disable nix-daemon.service

Reviewer 2: ```suggestion
sudo systemctl stop nix-daemon.service
sudo systemctl disable nix-daemon.socket nix-daemon.service
```

sudo systemctl daemon-reload
```

Then you can remove systemd service files:

```console

> sudo rm -f /etc/systemd/system/nix-daemon.service /etc/systemd/system/nix-daemon.socket

Reviewer 2: ```suggestion
sudo rm /etc/systemd/system/nix-daemon.service /etc/systemd/system/nix-daemon.socket
```

Author: `-f` is to make sure no error is emitted if the file does not exist. But since this is intended to run line-by-line by a human, maybe it's fine without `-f`.

Reviewer 2: Those files should exist if you installed the multi-user mode. Otherwise they might be somewhere else.

```

Also, the installer script uses systemd-tmpfiles (if presents) to create the socket directory. You may also want to remove the configuration for that:

```console

> sudo rm -f /etc/tmpfiles.d/nix-daemon.conf

Reviewer 2: ```suggestion
sudo rm /etc/tmpfiles.d/nix-daemon.conf
```
